src/uu/stty/src/stty.rs

```diff
@@ -245,6 +245,42 @@ fn print_terminal_size(termios: &Termios, opts: &Options) -> nix::Result<()> {
     Ok(())
 }
 
+fn control_char_to_string(mut cc: nix::libc::cc_t) -> nix::Result<String> {
+    let mut prefix = "";
+    if cc == 0 {
+        return Ok("<undef>".to_string());
+    } else if cc >= 0x80 {
+        cc &= !0x80;
+        prefix = "M-";
+    }
+
+    match cc {
+        0..=0x1f => Ok(format!("{}^{}", prefix, (b'@' + cc) as char)),
+        0x20..=0x7e => Ok(format!("{}{}", prefix, cc as char)),
+        0x7f => Ok(format!("{}^?", prefix)),
+        _ => Err(nix::errno::Errno::ERANGE),
+    }
```

Comment:
I tried to clean this up a bit. What do you think of this?
```suggestion
    if cc == 0 {
        return Ok("<undef>".to_string());
    }

    let (prefix, code) = if cc >= 0x80 {
        ("M-", cc - 0x80)
    } else {
        ("", cc)
    };

    let (ctrl, character) = match code {
        0..=0x1f => Ok(("^", (b'@' + code) as char)),
        0x20..=0x7e => Ok(("", code as char)),
        0x7f => Ok(("^", '?')),
        _ => Err(nix::errno::Errno::ERANGE),
    }?;

    Ok(format!("{prefix}{ctrl}{character}"))
```

Reply:
Sounds good, esp with `prefix` -> `meta_prefix` and `ctrl` -> `ctrl_prefix`

Reply:
Yes, that's even better!